### PR TITLE
Update golang.org/x/crypto to fix CVE-2020-29652

### LINKED ISSUE #103722: CVE-2020-29652: Update golang.org/x/crypto to v0.0.0-20201216223049-8b5274cf687f
#### What happened:

Kubernetes versions 1.19 and 1.20 are vulnerable to CVE-2020-29652

#### What you expected to happen:

Vulnerability to be mitigated.

#### How to reproduce it (as minimally and precisely as possible):

go.mod needs to be updated to include
```
golang.org/x/net => golang.org/x/net v0.0.0-20210520170846-37e1c6afe023
```
#### Anything else we need to know?:

This was fixed in the master and release-1.21 branches during the upgrade from go 1.15 to 1.16. However the older releases are still vulnerable.

/sig release

### PR DESCRIPTION
**Pull request by mengjiao-liu**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Kubernetes versions 1.20 are vulnerable to CVE-2020-29652.

Version v0.0.0-20201216223049-8b5274cf687f of golang.org/x/crypto fixes a vulnerability in the golang.org/x/crypto/ssh package which allowed clients to cause a panic in SSH servers.

An attacker can craft an authentication request message for the “gssapi-with-mic” method which will cause NewServerConn to panic via a nil pointer dereference if ServerConfig.GSSAPIWithMICConfig is nil.

This issue was discovered and reported by Joern Schneewesiz, GitLab Security Research Team, and is tracked as CVE-2020-29652.

Ref:
https://groups.google.com/g/golang-announce/c/ouZIlBimOsE?pli=1

CVE:
https://nvd.nist.gov/vuln/detail/CVE-2020-29652

#### Which issue(s) this PR fixes:

Fixes #103722

#### Special notes for your reviewer:
After updating crypto package, package golang.org/x/crypto/ssh/terminal is deprecated: this package moved to golang.org/x/term

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -86,7 +86,7 @@ require (
 	go.opentelemetry.io/otel/sdk v0.20.0
 	go.opentelemetry.io/otel/trace v0.20.0
 	go.opentelemetry.io/proto/otlp v0.7.0
-	golang.org/x/crypto v0.0.0-20210220033148-5ea612d1eb83
+	golang.org/x/crypto v0.0.0-20201216223049-8b5274cf687f
```
Comment (liggitt) [inline go.mod:89]: this is backleveling the library from 20210220033148 to 20201216223049

the current version should contain all prior fixes, right?

/hold

Comment (mengjiao-liu) [reply go.mod:89]: Oh, it looks like something is wrong. Let me check later.

**Comment by mengjiao-liu** [comment]
/remove-sig cli
/remove-sig apps
/remove-sig architecture
/remove-sig cloud-provider
/remove release-note-none
/remove-sig cluster-lifecycle
/remove-sig instrumentation
/remove-sig network
/remove-sig node
/remove-sig release
/remove-sig scheduling
/remove-sig storage
/remove-sig testing
/remove-wg structured-logging
/remove-area apiserver
/remove-area cloudprovider
/remove-area kubeadm
/remove-area kubectl
/remove-area kubelet
/remove-area/release-eng

File: go.mod (hunk 1 above)
Comment (mengjiao-liu) [reply go.mod:100]: You are right. 1.21 crypto package version contain the fix  and I should choose  apply this change to 1.20. Update.

**Comment by liggitt** [comment]
Kubernetes does not make use of the NewServerConn() method affected by CVE-2020-29652.

**Pull request closed without merging**